File: backEnd/restaurant/src/main/java/com/restaurant/dto/PurchaseRequest.java
package com.restaurant.dto;

import com.restaurant.model.form.Address;
import com.restaurant.model.form.Customer;
import com.restaurant.model.form.Item;
import com.restaurant.model.form.Request;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class PurchaseRequest {

    private Customer customer;
    private Request requestOrder;
    private List<Item> items = new ArrayList<>();
    private Address fromAddress;
    private Address toAddress;
}

File: frontEnd/restaurant-app/src/app/model/form/purchase-request.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Item } from "./item";
import { Request } from "./request";


export class PurchaseRequest {


  customer?: Customer;
  fromAddress?: Address;
  toAddress?: Address;
  requestOrder?: Request;
  items?: Item[];


}
